app/workers/digest_email_generation_worker.rb

```diff
@@ -1,13 +1,17 @@
 class DigestEmailGenerationWorker
   include Sidekiq::Worker
 
-  def perform(subscriber_id:, digest_run_id:)
-    @subscriber = Subscriber.find(subscriber_id)
-    @digest_run = DigestRun.find(digest_run_id)
+  def perform(digest_run_subscriber_id)
```

Comment:
Do we need to wrap this whole thing in a transaction in case it restarts? Or maybe at least the `generate_email_and_subscription_contents` method.

Reply:
That method did run in a transaction but I've expanded it a bit to wrap the queuing and marking complete bits. Added in ef4607c